public/app/features/dashboard-scene/settings/variables/utils.ts

```diff
@@ -0,0 +1,112 @@
+import { SelectableValue } from '@grafana/data';
+import {
+  AdHocFiltersVariable,
+  ConstantVariable,
+  CustomVariable,
+  DataSourceVariable,
+  IntervalVariable,
+  TextBoxVariable,
+  QueryVariable,
+} from '@grafana/scenes';
+import { VariableType } from '@grafana/schema';
+
+import { AdHocFiltersVariableEditor } from './editors/AdHocFiltersVariableEditor';
+import { ConstantVariableEditor } from './editors/ConstantVariableEditor';
+import { CustomVariableEditor } from './editors/CustomVariableEditor';
+import { DataSourceVariableEditor } from './editors/DataSourceVariableEditor';
+import { IntervalVariableEditor } from './editors/IntervalVariableEditor';
+import { QueryVariableEditor } from './editors/QueryVariableEditor';
+import { TextBoxVariableEditor } from './editors/TextBoxVariableEditor';
+
+interface EditableVariableConfig {
+  name: string;
+  description: string;
+  // FIXME: This should include all the scene objects
+  scene:
```

Comment:
I have to figure out the right type here because the type is any of the variable classes that correspond to variables. `variable` was kind of confusing because it is not the instance, it is the class itself. Maybe this can be uppercase since it is always a class, and we can call it `Variable`?